element_array_ephys/export/nwb/nwb.py

```diff
@@ -0,0 +1,637 @@
+import os
+import decimal
+import json
+import numpy as np
+import pynwb
+import datajoint as dj
+from element_interface.utils import find_full_path
+from hdmf.backends.hdf5 import H5DataIO
+from hdmf.data_utils import GenericDataChunkIterator
+from nwb_conversion_tools.tools.nwb_helpers import get_module
+from nwb_conversion_tools.tools.spikeinterface.spikeinterfacerecordingdatachunkiterator import (
+    SpikeInterfaceRecordingDataChunkIterator
+)
+from spikeinterface import extractors
+from tqdm import tqdm
+import warnings
+from ... import probe, ephys_acute
+
+assert probe.schema.is_activated(), 'probe not yet activated'
+
+for ephys in (ephys_acute,):
+    if ephys.schema.is_activated():
+        break
+else:
+    raise AssertionError('ephys not yet activated')
+
+
+class DecimalEncoder(json.JSONEncoder):
+    def default(self, o):
+        if isinstance(o, decimal.Decimal):
+            return str(o)
+        return super(DecimalEncoder, self).default(o)
+
+
+class LFPDataChunkIterator(GenericDataChunkIterator):
+    """
+    DataChunkIterator for LFP data that pulls data one channel at a time. Used when
+    reading LFP data from the database (as opposed to directly from source files)
+    """
+
+    def __init__(self, lfp_electrodes_query, chunk_length: int = 10000):
+        """
+        Parameters
+        ----------
+        lfp_electrodes_query: element_array_ephys.ephys_no_curation.LFP
+        chunk_length: int, optional
+            Chunks are blocks of disk space where data are stored contiguously and compressed
+        """
+        self.lfp_electrodes_query = lfp_electrodes_query
+        self.electrodes = self.lfp_electrodes_query.fetch("electrode")
+
+        first_record = (
+            self.lfp_electrodes_query & dict(electrode=self.electrodes[0])
+        ).fetch(as_dict=True)[0]
```

Comment:
```suggestion
        first_record = (
            self.lfp_electrodes_query & dict(electrode=self.electrodes[0])
        ).fetch1(as_dict=True)
```

Reply:
won't this fail if there is more than one result? I just want the first result, even if there are many. Actually, I think we can do that with `next`, right?

Reply:
If I am interpreting this function and the `add_ephys_lfp_from_dj_to_nwb` function correctly, `lfp_electrodes_query` is restricted to a specific `subject`, `session_datetime`, and `insertion_number`, and so by further restricting it to the `electrode` there should only be one entry resulting from this `fetch`.

Reply:
ok, good point